src/renderer/components/dialog-add-version.tsx

```diff
@@ -116,15 +131,23 @@ export class AddVersionDialog extends React.Component<
   }
 
   get buttons() {
-    const canSubmit = this.state.isValidElectron && this.state.isValidVersion;
+    const {
+      isValidElectron,
+      isValidVersion,
+      existingLocalVersion,
+    } = this.state;
+    const canAdd = isValidElectron && isValidVersion && !existingLocalVersion;
+    const canSwitch = isValidElectron && existingLocalVersion;
+    // swap to old local electron version if the user adds a new one with the same path
+    const shouldSetToPreviousVersion = isValidElectron && existingLocalVersion;
```

Comment:
`shouldSetToPreviousVersion` is a little prolix and its relationship to the other fields isn't clear at-a-glance. In particular someone who hasn't read this PR might not be able to tell if 'previous version' is the same thing as 'existing local version'.

It's also identical to the value of `canSwitch` so we probably don't need both.

There's real value in being able to understand code at-a-glance:

> `{text=canSwitch ? 'Switch' : 'Add'}`

Is something a developer can read and understand as soon as their eyes see it. It's short, it's direct, and it follows the hints the rest of the code has with use of the word 'switch', e.g. it's easy to understand the relationship between the two uses of 'switch' in that line.

> `shouldSetToPreviousVersion`

is slower to read because it takes longer to parse the words in the name and understand how they relate to the rest of the code.

```suggestion
```

Reply:
Thanks for this feedback, I'll keep this in mind for the future <3